radix-engine/src/track/db_key_mapper.rs

```diff
@@ -0,0 +1,88 @@
+use radix_engine_common::data::scrypto::{scrypto_decode, ScryptoDecode};
+use radix_engine_interface::crypto::{hash, Hash};
+use radix_engine_interface::types::{ModuleId, NodeId, SubstateKey};
+use radix_engine_store_interface::interface::{
+    DbPartitionKey, DbSortKey, PartitionEntry, SubstateDatabase,
+};
+use sbor::rust::prelude::*;
+
+/// A mapper between the business RE Node / Module / Substate IDs and database keys.
+pub trait DatabaseKeyMapper {
+    /// Converts the given RE Node and Module ID to the database partition's key.
+    fn to_db_partition_key(node_id: &NodeId, module_id: ModuleId) -> DbPartitionKey;
+
+    /// Converts the given Substate key to the database's sort key.
+    fn to_db_sort_key(key: &SubstateKey) -> DbSortKey;
+}
+
+/// A [`DatabaseKeyMapper`] tailored for databases which cannot tolerate long common prefixes
```

Comment:
Do you think it's worth mentioning that this is the "real" (protocol-enforced) key mapper that we're using for our protocol?

Reply:
It's the 7th time I regret that we have a trait for this. (without the trait, it would be obvious that precisely this logic is used for real in the protocol)...